File: app/src/main/java/com/lucagiorgetti/surprix/ui/mainfragments/catalog/setdetail/BaseSetDetailAdapter.java
package com.lucagiorgetti.surprix.ui.mainfragments.catalog.setdetail;

import androidx.recyclerview.widget.RecyclerView;

import java.util.List;

public abstract class BaseSetDetailAdapter<T extends RecyclerView.ViewHolder> extends RecyclerView.Adapter<T> {

    abstract void setSurprises(List<CollectionSurprise> surprises);
}

File: app/src/main/java/com/lucagiorgetti/surprix/ui/mainfragments/catalog/setdetail/BaseSetDetailAdapter.kt
package com.lucagiorgetti.surprix.ui.mainfragments.catalog.setdetail

import androidx.recyclerview.widget.RecyclerView

abstract class BaseSetDetailAdapter<T : RecyclerView.ViewHolder> : RecyclerView.Adapter<T>() {
    abstract fun setSurprises(surprises: List<CollectionSurprise>)
}
